### PR TITLE
shared_entities id field should be a UUID, not a integer

### LINKED ISSUE #3385: shared_entities id field should be a UUID, not a integer
There cannot be integer primary keys in any cartodb model related to dynamic data

### PR DESCRIPTION
**Pull request by Kartones**
Closes #3385

### CODE REVIEW

**Comment by Kartones** [comment]
@juanignaciosl CR please (tested in local & staging)

File: app/models/carto/shared_entity.rb [hunk 1]
```diff
@@ -2,14 +2,18 @@
...
   class SharedEntity < ActiveRecord::Base
     ENTITY_TYPE_VISUALIZATION = CartoDB::SharedEntity::ENTITY_TYPE_VISUALIZATION
 
-    scope :shared_visualizations, where(:entity_type => ENTITY_TYPE_VISUALIZATION)
+    scope :shared_visualizations, where(entity_type: ENTITY_TYPE_VISUALIZATION)
 
     belongs_to :visualization, foreign_key: :entity_id, inverse_of: :shared_entities
 
-  end
+    attr_accessible :entity_id, :recipient_id
```
Comment (juanignaciosl) [inline app/models/carto/shared_entity.rb]: Not a problem, but why is this needed?

**Comment by juanignaciosl** [comment]
Uhm, I must admit there're two points that feel unusual to me:
- Making a double column PK is better from a relational point of view but I think it will be the only one at CartoDB, and I haven't used Rails with double column PK.
- Column "desactivation" and column drop, strictly speaking, should be in separate deployments.

Anyway, if you have tried it, up to you :+1:

**Comment by Kartones** [comment]
I've tried and seems to work nicely, in fact we were already enforcing it via code ([see here](https://github.com/CartoDB/cartodb/blob/3385-removing-shared-entities-id/app/models/shared_entity.rb#L26)).

About the second point yep, you're right I deployed first the disabling then the migration, so I'm going to separate it just in case.

**Comment by juanignaciosl** [comment]
:+1:
